### PR TITLE
Curve448: Add 64-bit reference implementation

### PR DESCRIPTION
**Pull request by amitay**
This PR supersedes PR #14031.

The 64-bit reference code is resurrected from previously deleted code.  The 64-bit implementation is automatically selected if the compiler supports 128-bit integers, otherwise 32-bit implementation is used.

### CODE REVIEW

**Comment by paulidale** [comment]
Are the license issues mentioned in #13641 still relevant here?

**Comment by amitay** [comment]
Similar to PR #14031, this patch set resurrects deleted code which was already licensed.  PR #13641 re-added the deleted code (marking new attribution) and hence the license issue was raised.

**Comment by mattcaswell** [comment]
> I'm putting a license hold until this question is answered.

This resurrects code that was already in the repo, and was deleted - but the original submission was done under CLA and hence this resurrected code is still covered by the CLA. Therefore there is no licence issue here. Removing the hold.

**Review by mattcaswell — APPROVED** [review]
This is a fantastic piece of work.

File: Configure [hunk 1]
```diff
@@ -1573,6 +1573,20 @@ if (!$disabled{asm} && !$predefined_C{__MACH__} && $^O ne 'VMS') {
...
+    open(PIPE, "$cc -E -dM - </dev/null 2>&1 |");
+    while(<PIPE>) {
+        if (m/__SIZEOF_INT128__/) {
+            $config{use_int128} = 1;
+            last;
+        }
+    }
+    close(PIPE);
+}
+
```
Comment (mattcaswell) [inline Configure:1589]: As an aside (i.e. not for this PR), now that we  have this detection I think we could automatically enable `ec_nistp_64_gcc_128`

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed!! Thanks for the contribution and for sticking with this.

**Pull request closed without merging**